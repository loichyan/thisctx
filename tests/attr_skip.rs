use thisctx::WithContext;

#[derive(WithContext)]
enum Error {
    GenerateContext,
    #[thisctx(skip)]
    NotGenerateContext,
}

#[test]
fn attr_skip() {
    let _ = GenerateContext;
    let _ = Error::NotGenerateContext;
}
