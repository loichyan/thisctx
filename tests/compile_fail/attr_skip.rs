#[derive(thisctx::WithContextNext)]
#[thisctx(skip)]
enum Error {
    #[thisctx(skip = false)]
    GenerateContext,
    NotGenerateContext,
}

fn main() {
    let _ = GenerateContext;
    let _ = NotGenerateContext;
}
