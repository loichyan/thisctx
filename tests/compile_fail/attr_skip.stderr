error[E0425]: cannot find value `NotGenerateContext` in this scope
  --> tests/compile_fail/attr_skip.rs:11:13
   |
1  | #[derive(thisctx::WithContextNext)]
   |          ------------------------ similarly named unit struct `GenerateContext` defined here
...
11 |     let _ = NotGenerateContext;
   |             ^^^^^^^^^^^^^^^^^^
   |
help: a unit struct with a similar name exists
   |
11 |     let _ = GenerateContext;
   |             ~~~~~~~~~~~~~~~
help: consider importing this unit variant
   |
2  | use crate::Error::NotGenerateContext;
   |
